plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,521 @@
+                    device["ledStatus"] = True if device["ledStatus"] == "Enabled" else False
+
+                self.log("Response of Access Point Configuration: " + str(device), "INFO")
+                response = self.dnac.wireless.configure_access_points(**device)
+                self.log("Response of Access Point Configuration: " + str(response), "INFO")
+                all_response.append(dict(macAdress=device["macAddress"], response=response))
+            except Exception as e:
+                self.log("AP config update Error" + device["macAddress"] + str(e), "ERROR")
+
+        if len(all_response) > 0:
+            return all_response
+
+
+def main():
+    """ main entry point for module execution
+    """
+    # Basic Ansible type check or assign default.
+    accepoint_spec = {'dnac_host': {'required': True, 'type': 'str'},
+                    'dnac_port': {'type': 'str', 'default': '443'},
+                    'dnac_username': {'type': 'str', 'default': 'admin'},
+                    'dnac_password': {'type': 'str', 'no_log': True},
+                    'device_fields': {'required': True, 'type': 'str'},
+                    'ap_selected_field': {'required': True, 'type': 'str'},
+                    'dnac_verify': {'type': 'bool', 'default': 'True'},
+                    'dnac_version': {'type': 'str', 'default': '2.2.3.3'},
+                    'dnac_debug': {'type': 'bool', 'default': False},
+                    'dnac_log': {'type': 'bool', 'default': False},
+                    'dnac_log_level': {'type': 'str', 'default': 'WARNING'},
+                    "dnac_log_file_path": {"type": 'str', "default": 'dnac.log'},
+                    'config_verify': {'type': 'bool', "default": False},
+                    "dnac_log_append": {"type": 'bool', "default": True},
+                    'dnac_api_task_timeout': {'type': 'int', "default": 1200},
+                    'dnac_task_poll_interval': {'type': 'int', "default": 2},
+                    'config': {'required': True, 'type': 'list', 'elements': 'dict'},
+                    'validate_response_schema': {'type': 'bool', 'default': True}
+                }
+    module = AnsibleModule(
+        argument_spec=accepoint_spec,
+        supports_check_mode=True
+    )
+
+    ccc_network = Accesspoint(module)
+
+    # Check the Input file should not be empty config param
+    if len(module.params.get('config')) < 1:
+        module.fail_json(msg='Access Point Should not be Empty, You may forget to pass input.yml',
+                         **ccc_network.result)
+
+    ccc_network.validate_input(module.params).check_return_status()
+
+    # Getting the AP details by passing the Mac Address of the device
+    # Comparing input data with current AP configuration detail
+    final_config = ccc_network.get_have()
+
+    if final_config:
+        # Updating the final filtered data to the update AP information
+        ccc_network.get_want(final_config)
```

Comment:
we have to use get_diff_merged or get_diff_deleted function based on the state given in the playbook.

Reply:
inside get have compare_ap_cofig_with_inputdata function will take care of this

Reply:
No, get_have function is just to collect the data from Catalyst Center and playbook task will be performed only in get_diff_<state> function only based on state.